e2e/testcases/apiservice.bats

```diff
@@ -38,18 +56,24 @@ test_teardown() {
   wait::for -t 240 -- nomos::repo_synced
 }
 
-@test "importer and syncer resilient to bad APIService" {
+@test "importer and syncer resilient to flaky APIService" {
+  # Create an APIService without backend so that Config Sync DiscoveryClient will
+  # get service unavailable errors when trying to parse resource
```

Comment:
Is the service unavailable error visible to the user? Can we add a validation for that if possible?

If we don't add any validation before restoring the backend, it is basically the same as the first test case.